src/core/sock/sock-redirect.h

```diff
@@ -85,6 +89,26 @@
 #include <mutex>
 #include <set>
 
+#ifdef XLIO_STATIC_BUILD
+#define XLIO_SYMBOL(_func)                    xlio_##_func
+#define SYSCALL(_func, ...)                   ::_func(__VA_ARGS__)
+#define XLIO_CALL(_func, ...)                 xlio_##_func(__VA_ARGS__)
+#define SYSCALL_ERRNO_UNSUPPORTED(_func, ...) SYSCALL(_func, __VA_ARGS__)
+#define VALID_SYSCALL(_func)                  (true)
+#else
+#define XLIO_SYMBOL(_func) _func
+#if defined(__GNUC__) && !defined(__clang__)
+#define VALID_SYSCALL(_func) (__builtin_addressof(orig_os_api._func) != nullptr)
+#else
+#define VALID_SYSCALL(_func) ((orig_os_api._func) != nullptr)
+#endif
+#define SYSCALL(_func, ...)                                                                        \
```

Comment:
In case VALID_SYSCALL(_func) is true why should we call get_orig_funcs()? shouldn't be the opposite?

Reply:
Yes, change is done.